### DISCUSSION TITLE
createMenuEntry is marked as deprecated. What should be used instead?

### DISCUSSION THREAD

**Discussion by Dgarc359** [runelite/runelite#18506] [Q&A] [answered]
Hello,

I am working on a plugin, I was investigating how the ground markers plugin adds menu entries when a player shift clicks.

It [seems like it is leveraging](https://github.com/runelite/runelite/blob/05c56faf149011300038b034d731cd7ad3528c04/runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerPlugin.java#L258) `createMenuEntry` which is marked as [deprecated](https://static.runelite.net/api/runelite-api/net/runelite/api/Client.html#createMenuEntry(int))

The documentation does not say what is the recommended replacement for this method. What is the alternative for this, and when is the method planned to have support removed?

**Comment by Nightfirecat** [comment, marked as answer]
The documentation for that method has a "See also" section, which points to the replacements for this API.
